### PR TITLE
Add input signal attributes to output timeout signal. Close #4

### LINKED ISSUE #4: suggestion: add ability to enrich with last signal
This block could be made more versatile by including the last signal it received before timing out, with a check box to exclude that signal (current behavior). For example, if I pass a signal {{$pi: 3.14159}} to a timeout block, and it does not get another signal within its interval, it will output only {{$timeout: (datetime.timedelta)}}. I am suggesting a UI-configurable option to include last signal, so the output after a timeout looks like: {{$pi: 3.14159, $timeout: (datetime.timedelta)}}

### PR DESCRIPTION
**Pull request by hansmosh**

### CODE REVIEW

**Comment by hansmosh** [comment]
Instead of adding a flag like the issue suggested, I went with making every output signal include the last signal to be processed by the block.

**Comment by hansmosh** [comment]
Should any copying of the signal happen? Or should we expect the block router to handle copying?

File: signal_timeout_block.py [hunk 1]
```diff
@@ -51,11 +54,12 @@ def process_group(self, signals, key):
 
         # Lock around the individual group
         with self._jobs_locks[key]:
+            print(key)
```
Comment (mattdodge) [inline signal_timeout_block.py]: Whoopsie

File: signal_timeout_block.py [hunk 2]
```diff
@@ -51,11 +54,12 @@ def process_group(self, signals, key):
...
             # Cancel any existing timeout jobs, then reschedule them
             self._cancel_timeout_jobs(key)
             for interval in self.intervals:
                 self._schedule_timeout_job(
-                    key, interval.interval, interval.repeatable)
+                    signals[-1:][0], key, interval.interval, interval.repeatable)
```
Comment (mattdodge) [inline signal_timeout_block.py]: `signals[-1]` I think looks cleaner than `signals[-1:][0]`, is there a reason for the list thing?

File: signal_timeout_block.py [hunk 3]
```diff
@@ -63,13 +67,15 @@ def _cancel_timeout_jobs(self, key):
...
                                key, interval, repeatable))
         self._jobs[key][interval] = Job(
-            self._timeout_job, interval, repeatable, key, interval)
+            self._timeout_job, interval, repeatable, signal, key, interval)
 
-    def _timeout_job(self, key, interval):
+    def _timeout_job(self, signal, key, interval):
         """ Triggered when an interval times out """
-        self.notify_signals([Signal({'timeout': interval, 'group': key})])
+        signal.timeout = interval
```
Comment (mattdodge) [inline signal_timeout_block.py:78]: Does this work? I always thought we needed to `setattr` on signals

File: README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 SignalTimeout
 =======
 
-Notifies a timeout signal when no signals have been processed by this block for the defined intervals. A timeout signal has a *group* attribute that specifies the group (default 'null') and a *timeout* attribute that is a python datetime.timedelta specifying the configured interval that triggered the signal.
+Notifies a timeout signal when no signals have been processed by this block for the defined intervals. A timeout signal is the last signal to enter the block with and added *group* attribute that specifies the group (default 'null') and a *timeout* attribute that is a python datetime.timedelta specifying the configured interval that triggered the signal.
```
Comment (mattdodge) [inline README.md]: "with an\* added"

File: signal_timeout_block.py [hunk 5]
```diff
@@ -33,7 +36,7 @@ class SignalTimeout(GroupBy, Block):
     """
 
     intervals = ListProperty(Interval, title='Timeout Intervals')
-    version = VersionProperty('1.0.0')
+    version = VersionProperty('1.1.0')
```
Comment (mattdodge) [inline signal_timeout_block.py]: This might warrant a version 2, what do you think?

Comment (hansmosh) [inline signal_timeout_block.py]: It's definitely close to being a 2.0.0.  The block was documented as having an output signal with 'timeout' and 'group' attributes. Now it has those two attributes plus more. I suppose that is enough to potentially break backwards compatibility in some services..

File: signal_timeout_block.py (hunk 3 above)
Comment (hansmosh) [inline signal_timeout_block.py:78]: We do tend to use setattr but this appears to be working in the unittests

```
If the variable (attribute) name is known at "compile" time (hard-coded):
   e.foo = 1
If the variable name is determined at runtime:
   setattr(e, name, value)
```

source: http://rosettacode.org/wiki/Add_a_variable_to_a_class_instance_at_runtime#Python

Comment (mattdodge) [inline signal_timeout_block.py:78]: Oh, sweet, good to know!

**Pull request merged**